### PR TITLE
docdb input bindings

### PR DESCRIPTION
**Pull request by brettsam**
Adds DocDB input bindings to support the following parameter signatures:

**Item Input Binding**
With these, any updates to the doc will be persisted by calling `ReplaceDocumentAsync` when the method exits:
`[DocumentDB("ItemDb", "ItemCollection", Id = "{documentId}")] Document doc`
`[DocumentDB("ItemDb", "ItemCollection", Id = "{documentId}")] AnyPoco doc`
`[DocumentDB("ItemDb", "ItemCollection", Id = "{documentId}")] JObject doc`

**Client Input Binding**
If you need a full DocumentClient:
`[DocumentDB] DocumentClient client`

### CODE REVIEW

File: src/WebJobs.Extensions.DocumentDB/Bindings/CompositeBindingProvider.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    // TODO: Revisit this pattern to remove the need for this provider.
+    internal class CompositeBindingProvider : IBindingProvider
+    {
```
Comment (brettsam) [inline src/WebJobs.Extensions.DocumentDB/Bindings/CompositeBindingProvider.cs]: Copying my other pattern for beta was faster than coming up with something new -- but now we have at least two bindings doing the same thing. Already logged as #48

File: src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBItemValueBinder.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        //{
+        //    try
+        //    {
+        //        action();
+        //    }
+        //    catch (AggregateException)
+        //    {
+        //        // TODO: needed?
+        //    }
+        //}
```
Comment (paulbatum) [inline src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBItemValueBinder.cs]: Dead code

**Comment by paulbatum** [comment]
Looks good to me. Is there going to be much corresponding work on the script repo?

File: src/WebJobs.Extensions.DocumentDB/Services/IDocumentDBService.cs [hunk 3]
```diff
@@ -4,13 +4,14 @@
...
 using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
 
 namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
 {
     /// <summary>
     /// An abstraction layer for communicating with a DocumentDB account.
     /// </summary>
-    public interface IDocumentDBService
+    internal interface IDocumentDBService
```
Comment (mathewc) [inline src/WebJobs.Extensions.DocumentDB/Services/IDocumentDBService.cs]: Please add a test to PublicSurfaceTests for this binding, and review your surface area

**Comment by mathewc** [comment]
I took a quick look, and made a comment. Other than that, looks good.

File: src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBItemValueBinder.cs (hunk 2 above)
Comment (brettsam) [inline src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBItemValueBinder.cs]: Done.

File: src/WebJobs.Extensions.DocumentDB/Services/IDocumentDBService.cs (hunk 3 above)
Comment (brettsam) [inline src/WebJobs.Extensions.DocumentDB/Services/IDocumentDBService.cs]: Nice, I hadn't seen those. Added EasyTables as well.

**Comment by brettsam** [comment]
@paulbatum -- there won't be a ton of work in Scripts. But I'll get to work on that ASAP.

**Pull request merged**